arviz/plots/backends/matplotlib/traceplot.py

```diff
@@ -219,6 +225,10 @@ def plot_trace(
                 line_values = [vlines]
             else:
                 line_values = np.atleast_1d(vlines).ravel()
+                if not all(np.issubdtype(type(val), np.number) for val in line_values):
```

Comment:
line_values will be a 1d numpy array here, `np.issubdtype(line_values, np.number)` should work

Reply:
`np.issubdtype` works only on individual datatypes, following code is throwing the error:
```
line_values = np.atleast_1d("hi").ravel()
if np.issubdtype(line_values, np.number):
    print("Failed")
```
Error: `TypeError: data type not understood`. 
We need to check for the individual elements.

Reply:
`np.issubdtype(line_values.dtype, np.number)` np.issubdtype expects a dtype, not an array, forgot about that

Reply:
Yes, it'll work for same datatypes array `np.issubdtype(line_values.dtype, np.number)`. Can  it be case, if line_position is array of different datatypes say `[1, 3, "hi"]`!

Reply:
numpy arrays have one dtype, and cast all inputs to this dtype. Therefore, as numbers can always be cast to str but not the other way around, the dtype would be string/unicode... something like this

Reply:
Okay, thanks for the clarification :)